src/main/java/org/jabref/logic/importer/fileformat/CffImporter.java

```diff
@@ -92,73 +103,99 @@ public CffIdentifier() {
-        String authorStr = citation.authors.stream()
-                        .map(author -> author.values)
-                        .map(vals -> vals.get("name") != null ?
-                                new Author(vals.get("name"), "", "", "", "") :
-                                new Author(vals.get("given-names"), null, vals.get("name-particle"),
-                                        vals.get("family-names"), vals.get("name-suffix")))
-                        .collect(AuthorList.collect())
-                        .getAsFirstLastNamesWithAnd();
-        entryMap.put(StandardField.AUTHOR, authorStr);
-
-        // Select DOI to keep
-        if ((entryMap.get(StandardField.DOI) == null) && (citation.ids != null)) {
-            List<CffIdentifier> doiIds = citation.ids.stream()
-                            .filter(id -> "doi".equals(id.type))
-                            .collect(Collectors.toList());
-            if (doiIds.size() == 1) {
-                entryMap.put(StandardField.DOI, doiIds.getFirst().value);
-            }
+        StandardEntryType entryType = StandardEntryType.Misc;
+        if (citation.type != null) {
+            entryType = mapType(citation.type);
         }
+        BibEntry entry = new BibEntry(entryType);
+        HashMap<Field, String> entryMap = new HashMap<>();
 
-        // Select SWHID to keep
-        if (citation.ids != null) {
-            List<String> swhIds = citation.ids.stream()
-                                           .filter(id -> "swh".equals(id.type))
-                                           .map(id -> id.value)
-                                           .collect(Collectors.toList());
-
-            if (swhIds.size() == 1) {
-                entryMap.put(BiblatexSoftwareField.SWHID, swhIds.getFirst());
-            } else if (swhIds.size() > 1) {
-                List<String> relSwhIds = swhIds.stream()
-                                               .filter(id -> id.split(":").length > 3) // quick filter for invalid swhids
-                                               .filter(id -> "rel".equals(id.split(":")[2]))
-                                               .collect(Collectors.toList());
-                if (relSwhIds.size() == 1) {
-                    entryMap.put(BiblatexSoftwareField.SWHID, relSwhIds.getFirst());
-                }
-            }
+        if (citation.getPreferredCitation() != null) {
+            PreferredCitationMethod(citation.getPreferredCitation(), entryMap, entry);
         }
 
-        BibEntry entry = new BibEntry(entryType);
-        entry.setField(entryMap);
+        MainCffContentMethod(citation, entryMap, entry);
+        entryMap.forEach(entry::setField);
 
         List<BibEntry> entriesList = new ArrayList<>();
         entriesList.add(entry);
 
         return new ParserResult(entriesList);
     }
 
+    private void PreferredCitationMethod(JsonNode preferredCitation, Map<Field, String> entryMap, BibEntry entry) {
```

Comment:
method start with a lower case letter (lower camel case)

Reply:
Sure I'll change the method name. Thank you!